### DISCUSSION TITLE
How to save new entity with nested entities included

### DISCUSSION THREAD

**Discussion by andreystavitsky** [flutterdata/flutter_data#224] [Q&A] [answered]
For example, I have the Place model with
final HasMany<PlacePhoto> photos;

I need to save newly created Place in my JSON API. There's a requirement that photos must contain at least 2 photo. How to send request to API with body containing nested PlacePhoto entities?

As I remember, there's no implementation for this, and I need to override the serialize method of model? 

```
final Place myPlace = Place (
          ...
          photos: HasMany());

      for (var photo in currentState.photos) {
        final newPhoto = PlacePhoto(
            bucketUrl: photo.bucketPath,
            url: photo.uploadUrl,
            helpRequest: BelongsTo(myPlace));

        myPlace.photos.add(newPhoto);
      }
```

I'm trying to serialize that model, but photos array is empty.

**Comment by frank06** [comment, marked as answer]
The default will send IDs so if your model doesn't have an ID it won't send anything. If you need to send nested models override the default serializer. In your place adapter:

```dart
  @override
  Future<Map<String, dynamic>> serialize(T model,
      {bool withRelationships = true}) async {
    final map = await super.serialize(model, withRelationships: true);
    map['photos'] = mapIdsIntoModels(map['photos']);
    return map;
  }
```

To map IDs into models you can use the finders for example `await photoAdapter.findOne(photoId, remote: false)` and then turn them into maps via `photoAdapter.serialize(model)`.

To get the photo adapter inside your place adapter, look it up in `adapters`

**Reply by andreystavitsky** [reply]
so I need to save PlacePhotos locally first, right?

**Reply by frank06** [reply]
I think so yes
